dev/stats/get_important_pr_candidates.py

```diff
@@ -0,0 +1,187 @@
+
+logger = logging.getLogger(__name__)
+
+console = Console(width=400, color_system="standard")
+
+option_github_token = click.option(
+    "--github-token",
+    type=str,
+    required=True,
+    help=textwrap.dedent(
+        """
+        GitHub token used to authenticate.
+        You can set omit it if you have GITHUB_TOKEN env variable set
+        Can be generated with:
+        https://github.com/settings/tokens/new?description=Read%20issues&scopes=repo:status"""
+    ),
+    envvar='GITHUB_TOKEN',
+)
+
+PROVIDER_SCORE = 0.5
+REGULAR_SCORE = 1.0
+
+REVIEW_INTERACTION_VALUE = 1.0
+COMMENT_INTERACTION_VALUE = 1.0
+REACTION_INTERACTION_VALUE = 0.1
+
+
+class PrStat:
+    def __init__(self, pull_request: PullRequest):
+        self.pull_request = pull_request
+        self._users: Set[str] = set()
+
+    @cached_property
+    def label_score(self) -> float:
+        for label in self.pull_request.labels:
+            if "provider" in label.name:
+                return PROVIDER_SCORE
+        return REGULAR_SCORE
+
+    @cached_property
+    def num_interactions(self) -> float:
+        interactions = 0.0
+        for comment in self.pull_request.get_comments():
+            interactions += COMMENT_INTERACTION_VALUE
+            self._users.add(comment.user.login)
+            for _ in comment.get_reactions():
+                interactions += REACTION_INTERACTION_VALUE
+        for review in self.pull_request.get_reviews():
+            interactions += REVIEW_INTERACTION_VALUE
+            self._users.add(review.user.login)
+        return interactions
+
+    @cached_property
+    def num_interacting_users(self) -> int:
+        _ = self.num_interactions  # make sure the _users set is populated
+        return len(self._users)
+
+    @cached_property
+    def num_changed_files(self) -> float:
+        return self.pull_request.changed_files
+
+    @cached_property
+    def score(self):
+        return (
+            1.0
+            * self.num_interactions
+            * self.label_score
+            * self.num_interacting_users
+            / (math.log10(self.num_changed_files) if self.num_changed_files > 10 else 1.0)
+        )
+
+    def __str__(self) -> str:
+        return (
+            f"Score: {self.score:.2f}: PR{self.pull_request.number} by @{self.pull_request.user.login}: "
+            f"`{self.pull_request.title}. "
+            f"Merged at {self.pull_request.merged_at}: {self.pull_request.html_url}"
+        )
+
+
+DAYS_BACK = 5
+# Current (or previous during first few days of the next month)
+DEFAULT_BEGINNING_OF_MONTH = (date.today() - timedelta(days=DAYS_BACK)).replace(day=1)
+DEFAULT_END_OF_MONTH = (
+    deepcopy(DEFAULT_BEGINNING_OF_MONTH)
+    .replace(month=1 if DEFAULT_BEGINNING_OF_MONTH == 12 else DEFAULT_BEGINNING_OF_MONTH.month + 1)
+    .replace(
+        year=DEFAULT_BEGINNING_OF_MONTH.year + 1
+        if DEFAULT_BEGINNING_OF_MONTH.month == 12
+        else DEFAULT_BEGINNING_OF_MONTH.year
+    )
+)
```

Comment:
Lets use pendulum?

```suggestion
DEFAULT_BEGINNING_OF_MONTH = pendulum.now().start_of('month")
DEFAULT_END_OF_MONTH = DEFAULT_BEGINNING_OF_MONTH.add(months=1)
```

Reply:
I try to avoid adding new dependencies to "dev scripts" because they make necessary to update "dev/requirements.txt" generally update the viratualenv.  We do not have pendulum yet in there. Do you think it's worth it :) ?